provider/pkg/provider/helm_release.go

```diff
@@ -353,19 +357,23 @@ func (r *helmReleaseProvider) Check(ctx context.Context, req *pulumirpc.CheckReq
 			return nil, err
 		}
 
-		resourceNames, err := r.computeResourceNames(new, r.clientSet)
+		resourceInfo, err := r.computeResourceInfo(new, r.clientSet)
 		if err != nil && errors.Is(err, fs.ErrNotExist) {
 			// Likely because the chart is not readily available (e.g. import of chart where no repo info is stored).
 			// Declare bankruptcy in being able to determine the underlying resources and hope for the best
 			// further down the operations.
-			resourceNames = nil
+			resourceInfo = nil
 		} else if err != nil {
 			return nil, err
 		}
 
-		if len(new.ResourceNames) == 0 {
-			new.ResourceNames = resourceNames
+		if resourceInfo != nil {
```

Comment:
Will `r.computeResourceInfo` ever return `resourceInfo = nil` and `err = nil`? If not, then I think this `if` check would be clearer as the `else` block on the previous chain.